### PR TITLE
Add Integration test and CI

### PR DESCRIPTION
**Pull request by bhamail**
adds an integration test and associated "reference" files that must match the output generated by the integration test.

Also has an initial Travis CI config file to ensure shiny new tests are run.

remove 'spock' dependency

### CODE REVIEW

File: src/test/resources/projects/it1/reference/pom.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Sonatype Nexus (TM) Open Source Version
+    Copyright (c) 2018-present Sonatype, Inc.
```
Comment (DarthHater) [inline src/test/resources/projects/it1/reference/pom.xml]: A note: all of these years are wrong (it should be 2019). That's likely just something we need to fix in the archetype and unrelated to this, but worth pointing out.

File: src/test/resources/projects/it1/reference/pom.xml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    Eclipse Foundation. All other trademarks are the property of their respective owners.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.sonatype.nexus.plugins</groupId>
+    <artifactId>nexus-plugins</artifactId>
+    <version>3.14.0-04</version>
```
Comment (DarthHater) [inline src/test/resources/projects/it1/reference/pom.xml]: Another note: we should update this fairly soon to newest version of Nexus Repo (3.15.0-01 in this case).

File: src/test/resources/projects/it1/reference/pom.xml (hunk 1 above)
Comment (bhamail) [reply src/test/resources/projects/it1/reference/pom.xml]: Cool. I will use this as a chance to intentionally break the IT suite, and verify CI. :)

File: src/test/resources/projects/it1/reference/pom.xml (hunk 2 above)
Comment (bhamail) [reply src/test/resources/projects/it1/reference/pom.xml]: Will do.

**Pull request merged**